### DISCUSSION TITLE
C/C++ 项目如何指定 xxx.dll，使用 add_links("xxx")，默认是提示 xxx.lib 找不到

### DISCUSSION THREAD

**Discussion by shuaiplusplus** [xmake-io/xmake#5127] [Q&A]
C/C++ 项目如何指定 xxx.dll，使用 add_links("xxx")，默认是提示 xxx.lib 找不到

**Comment by star-hengxing** [comment]
Windows 动态库需要 .lib 导入符号

可以用 add_links 添加 绝对路径，或者用 add_linkdirs 和 add_links 配合

配置后用 xmake -v 看编译参数

**Reply by shuaiplusplus** [reply]
windows上生成动态库的使用 xmake 的工程只生成了 dll，没有生成 .lib
```lua
target("dll_test")
    set_kind("shared")
    add_files("src/*.cpp")
target_end()
```
然后使用 xmake 命令编译，可以生成 dll_test.dll，但是没有找到你说的 lib

**Reply by star-hengxing** [reply]
msvc 编译动态库默认不导出符号
导出符号才会生成 .lib
可以偷懒用构建系统帮你导出符号：https://xmake.io/#/manual/custom_rule?id=utilssymbolsexport_all
